baseapp/baseapp.go

```diff
@@ -838,13 +849,21 @@ func makeABCIData(msgResponses []*codectypes.Any) ([]byte, error) {
 	return proto.Marshal(&sdk.TxMsgData{MsgResponses: msgResponses})
 }
 
-func createEvents(events sdk.Events, msg sdk.Msg) sdk.Events {
+func createEvents(cdc codec.Codec, events sdk.Events, msg sdk.Msg, msgV2 protov2.Message) sdk.Events {
 	eventMsgName := sdk.MsgTypeURL(msg)
 	msgEvent := sdk.NewEvent(sdk.EventTypeMessage, sdk.NewAttribute(sdk.AttributeKeyAction, eventMsgName))
 
 	// we set the signer attribute as the sender
-	if len(msg.GetSigners()) > 0 && !msg.GetSigners()[0].Empty() {
-		msgEvent = msgEvent.AppendAttributes(sdk.NewAttribute(sdk.AttributeKeySender, msg.GetSigners()[0].String()))
+	signers, err := cdc.GetMsgV2Signers(msgV2)
+	if err != nil {
+		panic(err)
+	}
+	if len(signers) > 0 && signers[0] != nil {
+		addrStr, err := cdc.InterfaceRegistry().SigningContext().AddressCodec().BytesToString(signers[0])
```

Comment:
nit: It seems kind of odd that address codecs are buried in interfaceRegistry.SigningContext.AddressCodec as they have us outside of signing.  Not sure what the alternative here is though besides refactoring `codec.Codec`, so no action needed.

Reply:
We could do some embedding maybe so that the call is not so long